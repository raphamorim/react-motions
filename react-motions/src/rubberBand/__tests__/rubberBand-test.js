import React from 'react';
import { mount } from 'enzyme';
import RubberBand from '../rubberBand';

describe('RubberBand', () => {
  it('should render', () => {
    const wrapper = shallow(<RubberBand>RubberBand</RubberBand>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should receive duration prop', () => {
    const wrapper = mount(<RubberBand duration={3}>RubberBand</RubberBand>);

    expect(wrapper.props().duration).toEqual(3);
  });
  it('should receive fillMode prop', () => {
    const wrapper = mount(<RubberBand fillMode="both">RubberBand</RubberBand>);
    expect(wrapper.props().fillMode).toEqual("both");
  });
  it('should render correctly when infinite', () => {
    const wrapper = mount(<RubberBand infinite>RubberBand</RubberBand>);

    expect(wrapper.props().infinite).toBe(true);
  });
});
