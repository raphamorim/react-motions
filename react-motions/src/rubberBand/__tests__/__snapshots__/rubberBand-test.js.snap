// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withRubberBand should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    style={
      Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "rubberBand",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "rubberBand",
      }
    }
  >
    <div>
      rubberBand
    </div>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        rubberBand
      </div>,
      "style": Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "rubberBand",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "rubberBand",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "rubberBand",
      },
      "ref": null,
      "rendered": "rubberBand",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          rubberBand
        </div>,
        "style": Object {
          "WebkitAnimationDuration": "1s",
          "WebkitAnimationFillMode": "both",
          "WebkitAnimationName": "rubberBand",
          "animationDuration": "1s",
          "animationFillMode": "both",
          "animationName": "rubberBand",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "rubberBand",
        },
        "ref": null,
        "rendered": "rubberBand",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
